### PR TITLE
Do not package tests with tenacity

### PR DESCRIPTION
**Pull request by penguinolog**
* move tests outside of package
* fix DeprecationWaring for `unittest` API (in tests)
* Do not raise exceptions classes, but raise exception instances

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -12,7 +12,7 @@ trim_trailing_whitespace = true
 indent_size = 4
 max_line_length = 120
 
-[*.ini]
+[*.{ini,cfg}]
```
Comment (penguinolog) [inline .editorconfig:15]: figured, that current `setup.cfg` is indented with 4 spaces -> do not reformat

File: tenacity/_asyncio.py [hunk 2]
```diff
@@ -58,7 +58,7 @@ async def __anext__(self):
         while True:
             do = self.iter(retry_state=self._retry_state)
             if do is None:
-                raise StopAsyncIteration
+                raise StopAsyncIteration()
```
Comment (jd) [inline tenacity/_asyncio.py:61]: unrelated?

**Review by jd — CHANGES_REQUESTED** [review]

File: tenacity/_asyncio.py (hunk 2 above)
Comment (penguinolog) [reply tenacity/_asyncio.py:61]: Formally bug, but yes: unrelated. It should be separate change

**Review by and-semakin — APPROVED** [review]
Looks good!

**Review by jd — DISMISSED** [review]
There are a lot of unrelated changes. It'd really be great to split this in multiple PRs.

**Comment by penguinolog** [comment]
Removed all changes except move & fix imports

**Review by jd — APPROVED** [review]
